UiPath.FreeRdpClient/UiPath.FreeRdpClient.Tests/RdpClientTests.cs

```diff
@@ -149,6 +151,77 @@ public async Task ShouldConnectWithDifferentPort()
         await Host.WaitNoSession(connectionSettings);
     }
 
+    [Fact]
+    public async Task DisconnectCallbackShouldBeCalledWhenSessionIsDisconnected()
+    {
+        var onDisconnectCalled = false;
+        using var restoreDisconnectCallback = HijackOnDisconnect();
+
+        var callbackCalled = false;
+        var user = await Host.GivenUser();
+        var connectionSettings = user.ToRdpConnectionSettings(
+            disconnectCallback: () => { callbackCalled = true; });
+
+        // Connect
+        var connection = await Connect(connectionSettings);
+        var sessionId = await Host.FindSession(connectionSettings);
+
+        // Garbage collect
+        connection = null;
+        connectionSettings.DisconnectCallback = null;
+        GC.Collect();
+        GC.WaitForFullGCComplete();
+
+        // Disconnect
+        _wts.DisconnectSession(server: null, sessionId, wait: true);
+        await Host.WaitNoSession(connectionSettings);
+        // FreeRDP takes a while to call disconnect after we manually disconnect the session
+        await WaitFor.Predicate(() => onDisconnectCalled);
+
+        callbackCalled.ShouldBeTrue();

```

Comment:
I don't remember why do you need this Hijack.... couldn't you just wait for callbackCalled = true?

Reply:
I don't either, removed it.